pipenv/patched/piptools/utils.py

```diff
@@ -21,16 +22,20 @@
 def clean_requires_python(candidates):
     """Get a cleaned list of all the candidates with valid specifiers in the `requires_python` attributes."""
     all_candidates = []
+    py_version = parse_version(os.environ.get('PIP_PYTHON_VERSION', '.'.join(map(str, sys.version_info[:3]))))
     for c in candidates:
         if c.requires_python:
             # Old specifications had people setting this to single digits
             # which is effectively the same as '>=digit,<digit+1'
             if c.requires_python.isdigit():
                 c.requires_python = '>={0},<{1}'.format(c.requires_python, int(c.requires_python) + 1)
             try:
-                SpecifierSet(c.requires_python)
+                specifierset = SpecifierSet(c.requires_python)
             except InvalidSpecifier:
                 continue
+            else:
+                if not specifierset.contains(py_version):
```

Comment:
This will always be true, because `os.environ.get('PIP_PYTHON_VERSION') = 'False'`

Please merge #2437 and the CI will pass